src/predictor/gpu_predictor.cu

```diff
@@ -237,6 +237,19 @@ class GPUPredictor : public xgboost::Predictor {
     }
   }
 
+  void PredictionDeviceOffsets(size_t total_size, size_t batch_offset, size_t batch_size,
```

Comment:
i do not know how b3bdc4e stabilized the build by reduced the test data size. the number of rows inside the dmatrix that the test exercises are tiny, that removing and/or adding dmatrixes should not have made any difference. it is likely that the tests were run on slaves that have lesser gpu's that are masking issues.

Reply:
Agreed, not sure what the CI is doing.